ui/utils.py

```diff
@@ -170,3 +177,119 @@ def write_x509_files():
     """Write the x509 certificate and key to files"""
     write_to_file(settings.MIT_WS_CERTIFICATE_FILE, settings.MIT_WS_CERTIFICATE)
     write_to_file(settings.MIT_WS_PRIVATE_KEY_FILE, settings.MIT_WS_PRIVATE_KEY)
+
+
+def get_video_analytics(video_key):
+    """Get video analytics data from Google Analytics."""
+    ga_client = get_google_analytics_client()
+    ga_response = ga_client.reports().batchGet(
+        body=generate_google_analytics_query(video_key)).execute()
+    try:
+        return parse_google_analytics_response(ga_response)
+    except Exception as exc:
+        raise GoogleAnalyticsException(
+            'Could not parse analytics response') from exc
+
+
+def get_google_analytics_client():
+    """Gets a Google Analytics client.
+
+    Returns:
+        analytics_client: An analytics client
+    """
+    _check_files_exist([settings.GA_KEYFILE])
+    try:
+        credentials = ServiceAccountCredentials.from_json_keyfile_name(
+            settings.GA_KEYFILE)
+        analytics_client = build('analyticsreporting', 'v4',
+                                 credentials=credentials)
+        return analytics_client
+    except Exception as exc:  # pylint: disable=broad-except
+        raise GoogleAnalyticsException('Something went wrong with creating a'
+                                       'GoogleAnaltics client') from exc
+
+
+def generate_google_analytics_query(video_key):
+    """Generates a Google Analytics query.
+
+    Returns:
+        analytics_query: a query dict suitable to use as the body of an
+        analytics 'batchGet' request.
+    """
+    if not settings.GA_VIEW_ID:
+        raise RuntimeError("Missing 'settings.GA_VIEW_ID'.")
+    ga_view_id = settings.GA_VIEW_ID
+    # https://developers.google.com/analytics/devguides/reporting/core/v3/reference
+    START_DATE = '2005-01-01'
+    END_DATE = '9999-01-01'
```

Comment:
This will return all available data, which should be fine.  But it might be worth getting some clarification from @pdpinch on whether the graph should show all data by default, or just the last X days, and/or if the user should be able to specify a date range.

Reply:
leaving as-is for now (return all available data), since we can change this easily later.

Reply:
As an MVP approach, I think we should return all data. I'm not expecting a lot of traffic so in order for our users to make any inference from the data, they'll probably need all of it.